src/tests/functional/shared_test_classes/src/single_op/non_max_suppression.cpp

```diff
@@ -148,5 +148,251 @@ void Nms9LayerTest::SetUp() {
+    InputTypes input_types;
+    InputShapeParams input_shape_params;
+    int max_out_boxes_per_class;
+    float iou_thr, score_thr, soft_nms_sigma;
+    op::v5::NonMaxSuppression::BoxEncodingType box_encoding;
+    bool sort_res_descend;
+    ov::element::Type out_type;
+    std::tie(input_shape_params,
+             input_types,
+             max_out_boxes_per_class,
+             iou_thr,
+             score_thr,
+             soft_nms_sigma,
+             box_encoding,
+             sort_res_descend,
+             out_type,
+             targetDevice) = this->GetParam();
+
+    size_t num_batches, num_boxes, num_classes;
+    std::tie(num_batches, num_boxes, num_classes) = input_shape_params;
+
+    auto iouFunc = [](const Box& boxI, const Box& boxJ) {
+        const Rect& rectI = boxI.rect;
+        const Rect& rectJ = boxJ.rect;
+
+        float areaI = (rectI.y2 - rectI.y1) * (rectI.x2 - rectI.x1);
+        float areaJ = (rectJ.y2 - rectJ.y1) * (rectJ.x2 - rectJ.x1);
+
+        if (areaI <= 0.0f || areaJ <= 0.0f) {
+            return 0.0f;
+        }
+
+        float intersection_ymin = std::max(rectI.y1, rectJ.y1);
+        float intersection_xmin = std::max(rectI.x1, rectJ.x1);
+        float intersection_ymax = std::min(rectI.y2, rectJ.y2);
+        float intersection_xmax = std::min(rectI.x2, rectJ.x2);
+
+        float intersection_area = std::max(intersection_ymax - intersection_ymin, 0.0f) *
+                                  std::max(intersection_xmax - intersection_xmin, 0.0f);
+
+        return intersection_area / (areaI + areaJ - intersection_area);
+    };
+
+    // Get input bboxes' coords
+    std::vector<std::vector<Rect>> coordList(num_batches, std::vector<Rect>(num_boxes));
+    {
+        const auto& input = inputs[function->get_parameters().at(0)];
+        std::vector<double> buffer(input.get_size());
+        if (input.get_element_type() == ov::element::f32) {
+            convert(input.data<float>(), buffer.data(), input.get_size());
+        } else if (input.get_element_type() == ov::element::f16) {
+            convert(input.data<float16>(), buffer.data(), input.get_size());
+        } else {
+            const auto ptr = input.data<double>();
+            std::copy(ptr, ptr + input.get_size(), buffer.begin());
+        }
+
+        for (size_t i = 0; i < num_batches; ++i) {
+            for (size_t j = 0; j < num_boxes; ++j) {
+                const int32_t y1 = static_cast<int32_t>(buffer[(i * num_boxes + j) * 4 + 0]);
+                const int32_t x1 = static_cast<int32_t>(buffer[(i * num_boxes + j) * 4 + 1]);
+                const int32_t y2 = static_cast<int32_t>(buffer[(i * num_boxes + j) * 4 + 2]);
+                const int32_t x2 = static_cast<int32_t>(buffer[(i * num_boxes + j) * 4 + 3]);
+
+                coordList[i][j] = {std::min(y1, y2), std::min(x1, x2), std::max(y1, y2), std::max(x1, x2)};
+            }
+        }
+    }
+
+    auto compareBox = [](const Box& boxA, const Box& boxB) {
```

Comment:
For sort function is better to define less `operator<`,
the lambda should met requirement `!comp(a, b) && !comp(b, a)` if is not defined correctly e.g. in Debug build type using MSVC it can fail in compilation or execution as this can compiler can do additional checks.